modin/pandas/test/test_io.py

```diff
@@ -1279,6 +1279,15 @@ def wrapped_read_csv(file, method):
 
         df_equals(modin_df, pandas_df)
 
+    @pytest.mark.parametrize("read_mode", ["r", "rb"])
+    def test_read_csv_file_handle(self, read_mode, make_csv_file):
+        unique_filename = get_unique_filename()
+
+        make_csv_file(filename=unique_filename)
+
+        with open(unique_filename, mode=read_mode) as buffer:
+            eval_io(fn_name="read_csv", filepath_or_buffer=buffer)
```

Comment:
```suggestion
            df_pandas = pandas.read_csv(buffer)
            buffer.seek(0)
            df_modin = pd.read_csv(buffer)
            df_equals(df_modin, df_pandas)
```
I think here and in `test_read_json_file_handle` can be used direct pandas and modin results comparison, because after the first usage of file handle by `pandas.read_csv` in `eval_io`, file handle pointer is shifted and the second call of the same file handle in `pd.read_csv` causes `pandas.errors.EmptyDataError: No columns to parse from file` error

Reply:
Many thanks, that’s addressed the TeamCity tests. I bizzarely have the cloud test failing on one parametrised case of one test now... 🤔

Reply:
You are welcome! looks like `test cloud` test fails because file handle can't be serialized in the cloud mode and it should be converted to the path object as it was done in the `csv_dispatcher.py`. @anmyachev can you take a look and suggest which place this conversion should be done?

Reply:
We can do the same replacement here:
https://github.com/modin-project/modin/blob/5f03eb8b1e887f9f3b6990fb023576070d0e6cee/modin/data_management/factories/factories.py#L285

To do this, just check `a[0]`. 

Several suggestions:
 * It would be nice to make a warning that a filename will be used instead of a buffer;
 * try to move the actions of replacing the buffer to a filename into a separate function (remove duplicated code).

Reply:
Thanks very much guys, super helpful! I actually ended up leaving the cloud stuff for now and have added a skip to the pytest in when running with `--simulate-cloud=normal` because I think the issue there of handling non-picklable objects is a wider one that is coincidentally picked up by my tests, but perhaps not in scope for this PR but also because I couldn't immediately see a nice way to re-use the code I've defined in `text_file_dispatcher.py` in that factory file.

@anmyachev I've taken your feedback onboard and removed duplicate code by moving the functionality into the base class and then using `cls.get_path_from_buffer(...)` in the `csv,json,fwf` classes. Thanks for this, it does make it nicer and also made it easier to implement the warning that the filename was used instead of the buffer, since I could put it in that one place.

Reply:
@mzjp2 skipping cloud tests is ok for now. Good job!